include/lug/System/Logger.hpp

```diff
@@ -0,0 +1,116 @@
+#pragma once
+
+#include <memory>
+#include <set>
+#include <string>
+
+#include <lug/System/Export.hpp>
+#include <lug/System/Logger/Common.hpp>
+#include <lug/System/Logger/Handler.hpp>
+#include <lug/System/Logger/LoggingFacility.hpp>
+#include <lug/System/Logger/Message.hpp>
+#include <lug/System/Utils.hpp>
+
+namespace lug {
+namespace System {
+
+class LUG_SYSTEM_API Logger {
+public:
+    Logger(const std::string& loggerName);
+    virtual ~Logger() {}
```

Comment:
` = default` and do you really want to inherit from `Logger`? If not, just remove all virtual functions.